### PR TITLE
minor recreate refactor

### PR DESCRIPTION
**Pull request by enkore**
doesn't merge conflict with #1420

### CODE REVIEW

**Comment by enkore** [comment]
merge?

File: src/borg/archive.py [hunk 1]
```diff
@@ -1532,52 +1532,65 @@ def try_resume(self, archive, target_name):
...
-                    self.cache.chunk_incref(chunk.id, target.stats)
-                target.stats.nfiles += 1
-            target.add_item(item)
-        if item:
-            resume_from = item.path
+        last_old_item = self.copy_items(old_target, target)
+        if last_old_item:
+            resume_from = last_old_item.path
         else:
             resume_from = None
```
Comment (ThomasWaldmann) [inline src/borg/archive.py]: resume_from = last_old_item.path if last_old_item else None

File: src/borg/archive.py [hunk 2]
```diff
@@ -1532,52 +1532,65 @@ def try_resume(self, archive, target_name):
...
+    def copy_items(self, source_archive, target_archive):
+        item = None
+        for item in source_archive.iter_items():
+            if 'chunks' in item:
+                for chunk in item.chunks:
+                    self.cache.chunk_incref(chunk.id, target_archive.stats)
+                    target_archive.stats.nfiles += 1
+            target_archive.add_item(item)
+        if self.progress:
+            source_archive.stats.show_progress(final=True)  # XXX target_archive.stats?
```
Comment (ThomasWaldmann) [inline src/borg/archive.py]: something todo here?

**Comment by ThomasWaldmann** [comment]
one XXX found, other than that lgtm.

**Comment by enkore** [comment]
> Travis: An error occurred while generating the build script.

Hm. Restarting.

**Comment by ThomasWaldmann** [comment]
Test from Android.

**Pull request merged**